### ISSUE TITLE
postStartCommand fails: p: cannot create regular file '/workspace/bitcoin/.git/hooks/pre-push': No such file or directory

### ISSUE THREAD

**Issue by harding** [stickies-v/vs-core#2]
On 45c79dbfeb71a58340a4dcf5d494b8d702e5edcf install fails at:

```
Running the postStartCommand from devcontainer.json...

[514247 ms] Start: Run in container: /bin/sh -c /scripts/init.sh
Cloning into 'bitcoin'...
warning: templates not found in /home/harding/.config/git-template
remote: Enumerating objects: 258319, done.
remote: Counting objects: 100% (199/199), done.
remote: Compressing objects: 100% (141/141), done.
remote: Total 258319 (delta 117), reused 98 (delta 56), pack-reused 258120
Receiving objects: 100% (258319/258319), 231.10 MiB | 1.87 MiB/s, done.
Resolving deltas: 100% (185056/185056), done.
Renaming remote references: 100% (7/7), done.
cp: cannot create regular file '/workspace/bitcoin/.git/hooks/pre-push': No such file or directory
[649133 ms] postStartCommand failed with exit code 1. Skipping any further user-provided commands.
Done. Press any key to close the terminal.
```

Notes:

- It's unclear to me whether it's trying to create /workspace/bitcoin on my host filesystem on the container filesystem.  If it's the host system, I certainly don't have a /workspace directory.
- I don't think it's the issue here, but you can see it's trying to access my git templates.  The existence of those and other custom git configuration is a potential future source of failures in a command that expects git to behave consistently across systems.

**Comment by stickies-v** [comment]
> It's unclear to me whether it's trying to create /workspace/bitcoin on my host filesystem on the container filesystem.

It's on the container filesystem, `/workspace` is a named volume, see https://github.com/stickies-v/vs-core/blob/7288f4c767c9764d448c4b423a4c8c76ddae97d8/.devcontainer/docker-compose.yml#L10

You should be able to see this with `docker volume ls --filter name="_*workspace"` and remove it with `docker volume rm` to give you a clean slate for the setup process.

> I don't think it's the issue here, but you can see it's trying to access my git templates. 

Actually I think it might be the issue. The git template defines how a new repo is initialized, including the `hooks` subdir. Since no template is found (I'm not sure whether this is because you've customized your git templates, or because the setup process is just using the wrong git config), it seems like there's no `hooks` dir. 

Should be fixed in dc44672bfa47e32f2f61d11146a3e793208733c5

**Comment by harding** [comment]
dc44672bfa47e32f2f61d11146a3e793208733c5 works for me.  Thanks!  Closing as resolved.
